esp-bootloader-esp-idf/src/lib.rs

```diff
@@ -28,6 +28,10 @@ mod fmt;
 
 pub mod partitions;
 
+// thank you, OSX/mach-o
+#[cfg(all(feature = "ota", target_os = "none"))]
```

Comment:
Why are you building for macos?

Reply:
CI runs the host tests on OSX ... and that hates our metadata segment

Reply:
That can be helped, defmt also has a workaround for it. Not checking the module is less than optimal...

Reply:
mind pointing me to the defmt workaround (given we don't make ota non-optional ... but either then I'm interested in that workaround)